pkg/api/datasources.go

```diff
@@ -347,6 +347,17 @@ func validateJSONData(jsonData *simplejson.Json, cfg *setting.Cfg) error {
 			}
 		}
 	}
+
+	for key, value := range jsonData.Get("teamHeaders").MustMap() {
+		v := value.([]map[string]string)
+		for _, header := range v {
+			if header["header"] == cfg.AuthProxyHeaderName {
```

Comment:
This should probably reuse what we already have in regards to not forwarding auth headers, see https://github.com/grafana/grafana/blob/700e6e328799cb6ef4680fa233d713cbc696001b/pkg/util/proxyutil/reverse_proxy.go#L71

Reply:
fixed in 8ef005b7042c8197f2429a7c0e6256a75412b681